ntp-proto/src/system.rs

```diff
@@ -242,52 +242,48 @@ impl<SourceId: Hash + Eq + Copy + Debug, Controller: TimeSyncController<SourceId
         Ok(())
     }
 
-    #[allow(clippy::type_complexity)]
-    pub fn create_ntp_source(
+    pub fn create_source_controller(
```

Comment:
I don't think we should leak the controllers out of system. I'd much rather have a thin-ish wrapper for non-ntp sources as well. 

Although the idea of combining both nts and non-nts source creation isn't necessarily a bad thing.

Reply:
Should be fixed in cafc220 (as part of the other PR because here we do not have the sock source yet.)